core/block_rendering_rewrite/block_render_info.js

```diff
@@ -389,8 +392,10 @@ Blockly.BlockRendering.RenderInfo.prototype.getInRowSpacing_ = function(prev, ne
 
   // Spacing between a rounded corner and a previous or next connection
   if (prev.isRoundedCorner()){
```

Comment:
Would it make sense to change the squared corners so they're the same size as a rounded corner? It seems odd to have special spacing for one and not the other.

Also, this block is splitting two checks for prev.isSquareCorner() that should be combined.

Reply:
I just tried making squared corners have width (previously they had no width, and all width was taken up in the spacer instead).
The problem is that a square corner can come before a hat (in which case the corner should have zero width) or before a notch (in which case it should have non-zero width).  The corner's width should not depend on what comes before or after it, because that moves the spacing logic out of this single function.

I'll combine the isSquareCorner checks.

I think that once I start testing with more squared corners (stacked blocks) I'll have to change the square corner part of this as well, to have the slight difference between previous and next offsets.

Reply:
sgtm.